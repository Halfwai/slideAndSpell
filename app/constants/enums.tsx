export enum Direction {
    up = UP,
    down = DOWN,
    LEFT,
    RIGHT,
    FALSE
}