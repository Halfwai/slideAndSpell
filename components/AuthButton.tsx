import { TouchableOpacity, StyleSheet } from 'react-native';
import { useContext } from 'react';
import MyAppText from '@/components/MyAppText';
import { UserContext } from '@/utils/context';
import * as Haptics from 'expo-haptics';

interface AuthButtonProps {
    text: string;
    onPress: Function;
    style?: object;
}

export default function AuthButton(props: AuthButtonProps) {
    const userContext = useContext(UserContext);
    return (
        <TouchableOpacity 
            style={[styles.button, props.style]} 
            onPress={async () => {
                if (userContext && userContext.vibrate) {
                    await Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Light);
                }
                props.onPress()
            }}
            testID='authButton'
        >
            <MyAppText style={styles.buttonText}>{props.text}</MyAppText>
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    button: {
        padding: 12,
        borderRadius: 10,
        marginBottom: 20,
        alignItems: 'center',
        borderWidth: 1,
        width: '90%',
    }, 
    buttonText: {
        fontSize: 20,
        color: 'black',
        fontWeight: 'bold',
    },
})