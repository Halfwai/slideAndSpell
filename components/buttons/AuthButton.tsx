import React, { useContext } from 'react';
import { TouchableOpacity, StyleSheet } from 'react-native';
import * as Haptics from 'expo-haptics';
import { RFPercentage } from "react-native-responsive-fontsize";

// Import context
import { UserContext } from '@/utils/context';

// Import components
import MyAppText from '@/components/common/MyAppText';

// Setup props
interface AuthButtonProps {
    text: string;
    onPress: Function;
    style?: object;
    disabled?: boolean;
}

export default function AuthButton(props: AuthButtonProps) {
    // Get userContext
    const userContext = useContext(UserContext);
    return (
        <TouchableOpacity
            style={[styles.button, props.style]}
            onPress={async () => {
                if (props.disabled) return;
                // If vibrate settings are enabled vibrate on touch
                if (userContext && userContext.vibrate) {
                    await Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Light);
                }
                props.onPress()
            }}
            testID='authButton'
        >
            <MyAppText style={styles.buttonText}>{props.text}</MyAppText>
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    button: {
        padding: 12,
        borderRadius: 10,
        marginBottom: 20,
        alignItems: 'center',
        borderWidth: 1,
        width: '90%',
    },
    buttonText: {
        fontSize: RFPercentage(3),
        color: 'black',
        fontWeight: 'bold',
    },
})