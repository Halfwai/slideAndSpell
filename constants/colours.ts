// https://medium.com/@suraj27/global-colors-in-reactjs-custom-defined-3d4b76efa4fc

export const COLOURS = {
    blue: "#CBE6F7",
    green: "#8BF0BC",
    red: "#CB4C4E",
    grey: "#AAA"
};