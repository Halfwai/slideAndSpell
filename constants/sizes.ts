import { Dimensions } from 'react-native';


export const SIZES = {
    GAMEBOARD: Math.min(Dimensions.get("screen").height * 0.45, Dimensions.get("window").width * 0.85),
}