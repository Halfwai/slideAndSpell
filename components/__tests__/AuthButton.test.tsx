import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import AuthButton from '@/components/buttons/AuthButton';
import { COLOURS } from '@/constants/colours';

it(`Auth Button renders correctly with inherited style`, () => {
    const { getByTestId, toJSON } = render(
        <AuthButton text="Close" onPress={() => {}} style={{ backgroundColor: COLOURS.green }} />
    );
    const button = getByTestId('authButton');
    expect(toJSON()).toMatchSnapshot();
    expect(button).toHaveStyle({ backgroundColor: COLOURS.green });
});

it(`Auth Button press calls function`, () => {
    const testFunction = jest.fn()
    const { getByText, toJSON } = render(
        <AuthButton text="Close" onPress={() => {testFunction()}} style={{ backgroundColor: COLOURS.green }} />
    );
    const button = getByText('Close');
    fireEvent.press(button);
    expect(testFunction).toHaveBeenCalled();
});