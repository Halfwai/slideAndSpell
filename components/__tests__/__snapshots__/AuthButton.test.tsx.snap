// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Button renders correctly with inherited style 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#8BF0BC",
      "borderRadius": 10,
      "borderWidth": 1,
      "marginBottom": 20,
      "opacity": 1,
      "padding": 12,
      "width": "90%",
    }
  }
  testID="authButton"
>
  <Text
    style={
      [
        {
          "fontFamily": "Quicksand",
        },
        {
          "color": "black",
          "fontSize": 40,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Close
  </Text>
</View>
`;
